janus_core/helpers/mlip_calculators.py

```diff
@@ -116,223 +123,253 @@ def choose_calculator(
     # Fix torch 2.6 (must be before MLIP modules are loaded)
     _set_no_weights_only_load()
 
-    if arch == "mace":
-        from mace import __version__
-        from mace.calculators import MACECalculator
+    match arch:
+        case "mace":
+            from mace import __version__
+            from mace.calculators import MACECalculator
 
-        # No default `model`
-        if model is None:
-            raise ValueError(
-                f"Please specify `model`, as there is no default model for {arch}"
+            # No default `model`
+            if model is None:
+                raise ValueError(
+                    f"Please specify `model`, as there is no default model for {arch}"
+                )
+            # Default to float64 precision
+            kwargs.setdefault("default_dtype", "float64")
+
+            calculator = MACECalculator(model_paths=model, device=device, **kwargs)
+
+        case "mace_mp":
+            from mace import __version__
+            from mace.calculators import mace_mp
+
+            # Default to "small" model and float64 precision
+            model = model if model else "small"
+            kwargs.setdefault("default_dtype", "float64")
+
+            calculator = mace_mp(model=model, device=device, **kwargs)
+
+        case "mace_off":
+            from mace import __version__
+            from mace.calculators import mace_off
+
+            # Default to "small" model and float64 precision
+            model = model if model else "small"
+            kwargs.setdefault("default_dtype", "float64")
+
+            calculator = mace_off(model=model, device=device, **kwargs)
+
+        case "m3gnet":
+            from matgl import __version__, load_model
+            from matgl.apps.pes import Potential
+            from matgl.ext.ase import M3GNetCalculator
+            import torch
+
+            # Set before loading model to avoid type mismatches
+            torch.set_default_dtype(torch.float32)
+            kwargs.setdefault("stress_weight", 1.0 / 160.21766208)
+
+            # Use potential (from kwargs) if specified
+            # Otherwise, load the model if given a path, else use a default model
+            if isinstance(model, Potential):
+                potential = model
+                model = "loaded_Potential"
+            elif isinstance(model, Path):
+                if model.is_file():
+                    model = model.parent
+                potential = load_model(model)
+            elif isinstance(model, str):
+                potential = load_model(model)
+            else:
+                model = "M3GNet-MP-2021.2.8-DIRECT-PES"
+                potential = load_model(model)
```

Comment:
I think you could also rewrite this as a `match`:
```suggestion
            match model:
                case Potential():
                    potential = model
                    model = "loaded_Potential"
                case Path():
                    if model.is_file():
                        model = model.parent
                    potential = load_model(model)
                case str():
                    potential = load_model(model)
                case _:
                    model = "M3GNet-MP-2021.2.8-DIRECT-PES"
                    potential = load_model(model)
```